render_hawk_2/camera_class.cs

```diff
@@ -0,0 +1,71 @@
+class camera_class
+{
+    public double[] position;
+    public double[] vector_look;
+    public double[] vector_up;
+    public double[] vector_u;
+    public double[] vector_v;

```

Comment:
I remember deciding on `double[]` over a dedicated `Vector3D` class for the ability to loop over dimensions to reduce redundancy. I'd probably look into [operator overloading](https://msdn.microsoft.com/en-us/library/aa288467(v=vs.71).aspx) and utility functions to avoid both the redundancy and `for` loops.